File: orm/jkorm-mybatis/src/main/kotlin/net/jkcode/jkbenchmark/orm/mybatis/model/Employee.java
package net.jkcode.jkbenchmark.orm.mybatis.model;

public class Employee {
    private Integer id;
    private String title;
    private String email;
    private String gender;
    private Department dep;

    public Employee(){

    }

    public Employee(Integer id, String title, String email, String gender, Department dep) {
        this.id = id;
        this.title = title;
        this.email = email;
        this.gender = gender;
        this.dep = dep;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getGender() {
        return gender;
    }

    public void setGender(String gender) {
        this.gender = gender;
    }

    public Department getDep() {
        return dep;
    }

    public void setDep(Department dep) {
        this.dep = dep;
    }
}

File: orm/jkorm-mybatis/src/main/kotlin/net/jkcode/jkbenchmark/orm/jkorm/model/Employee.kt
package net.jkcode.jkbenchmark.orm.jkorm.model

import net.jkcode.jkmvc.orm.* 
import java.util.*

/**
 * 员工
 *
 * @author shijianhang<772910474@qq.com>
 * @date 2021-04-13 14:18:34
 */
class Employee(vararg pks: Any): Orm(*pks) {

	public constructor() : this(*arrayOf())

	// 伴随对象就是元数据
 	companion object m: OrmMeta(Employee::class, "员工", "employee", DbKeyNames("id")){
		init {
			belongsTo("dep", Department::class, "dep_id")
		}
	}

	// 代理属性读写
	public var id:Integer by property() //

	public var title:String by property() //  

	public var email:String by property() //  

	public var gender:String by property() //  

	public var depId:Integer by property() //

	public var dep:Department by property() // 从属于一个部门

}